### ISSUE TITLE
Works only without Zener diode?!?

### ISSUE THREAD

**Issue by TomSchimana** [No0ne/ps2pico#5]
Hi. 

I have assembled the ps2pico and have a question about it. I took the version which is for PS/2 Keyboards 

I currently have the whole thing put together on a breadboard. It didn't work initially. After some tests, I figured out to remove the Zener diode between GPIO14 and Ground. The other one from GPIO17 to Ground is still in place.

After removing it, everything worked immediately. Now I'm concerned that I might face issues with overvoltage later on.

In the picture, you can see my wild setup. 😉 
Inside the rectangle is the Zener diode from GPIO 17 to Ground, and where the arrow is pointing, I removed it. 
![ps2picotest](https://github.com/No0ne/ps2pico/assets/148782371/60ea7357-c381-40a0-b6ae-1e39221a5165)

On the VBUS, there is pretty much exactly 5 volts. If I measure GPIO 14 to Ground or even GPIO 17, it's 0.5?!?! Whether the Zener diode is connected or not. Maybe I'm doing something wrong or I assembled it incorrectly. The Zener diodes are 3.6V 5W.
When I place a 10K resistor between GPIO 14 and Ground, everything still works.

Should I just leave it or is something "wrong", even though it works without the diode?
Thanks in advance!

**Comment by No0ne** [comment]
I think the problem is that there are only 10k pull-ups on the board according to: http://land-boards.com/PiPicoMite01/PiPicoMite01_Rev2_Schematic.pdf
And the zener pulls down the voltage too much behind the second 10k.

But you could try the following:
use my other project https://github.com/No0ne/ps2x2pico/releases/tag/beta-0.8
and connect like this:
Board GND to Pico GND
Board 5V to Pico VBUS
Board J10 Pin 12 (=GP9) to Pico GPIO11
Board J10 Pin 11 (=GP8) to Pico GPIO12

This should work without any additional components.

**Comment by No0ne** [comment]
Here I've started another project for the picomite: https://github.com/No0ne/hid2cdc

**Comment by yugokoral** [comment]
Hello. First i am really sorry, because i really don't know where to ask... and i just got here borrowing this closed topic. I wonder if ps2x2pico works with Raspberry Pi Pico v2, which came out last year?

Well, i have a lot of retro computers, also i have some ps2 mice and kb. The main reason i want to build ps2x2pico is to convert my modern usb kb and mouse to ps2 protocol. Also i want to know more about PS/2 passthru. What is the purpose? Does that means, that i can have my ps2 mouse and kb connected together with my modern USB kb and mouse at the same time to the ps2 PC and that pico just reroute the PS2 inputs to PS2 outputs? Or do i have to switch somehow between USB or PS2 input, or does it works simultaneously out of the box?



Thank You very much for your help!!

**Comment by No0ne** [comment]
Hi, I've got a Pico 2 now, but havn't had time since christmas. Will test it this weekend.

The main reason for the passthru feature was the PiKVM integration, see: https://docs.pikvm.org/pico_hid/#ps2-keyboard-mouse
The USB input and PS/2 input work simultaneously.

**Comment by yugokoral** [comment]
I was trying to copy uf2 file to Pico 2 and than i realized that i have to compile a new uf2 file for Pico 2, because it is a different architecture. I got lost now... I am trying to compile your source code for Pico 2. I am new to programing and i am using Windows. I am struggling to get SDK 2.1.0 for windows and i can not find NMAKE program. I thought this is kinda easy peasy haha, i just got myself into a new rabbit hole :(

**Comment by No0ne** [comment]
I'll try to compile a pico2 binary as soo as I've some free time, maybe this weekend.

**Comment by No0ne** [comment]
Tried to compile a pico2 version but theres more stuff todo, not only setting the cpu to rp2350.
As my time is limited at the moment please run it on a pico1 or another rp2040 board.

**Comment by yugokoral** [comment]
Thank you. I really appreciate that you took time and tried it. It is a great work. Well, I will just wait for a new release, because I just bought four Picos 2, because i believed (thanks to all adverts), that it will work. I was a bit naive haha...

**Comment by yugokoral** [comment]
I just got pico from my coworker, to test it. Actually the device itself works pretty good. But i found two bugs. 

First one: When I have Y usb splitter connected to pico (and keyboard to the splitter, so i have another usb port for mouse), the NUM, CAPS and SCROL lock LEDs dont work. I tried splitter with HIDman, and it works, so there is no problem in splitter. 

Another problem is probably a delay problem. The PC wont boot, when pico is connected to my KVM switch as input device. It freezes before loading primary and secondary IDE ports. I think it's the same problem as in HIDman's case, which rasteri just solved a day ago. The pico responds to "led reset ??" command way too fast. Usually keyboards responds back with BAT code (0xAA) after cca 300ms. So hidman made a delay of 500ms and it works in HIDmans case. Here is a link of discussion:
 [](https://www.vogons.org/viewtopic.php?t=89777&start=460)
